packages/xml-views-validation/test/validators/element/unknown-tag-name-spec.ts

```diff
@@ -0,0 +1,525 @@
+import { partial } from "lodash";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { generateModel } from "@ui5-language-assistant/test-utils";
+import {
+  assertNoIssues as assertNoIssuesBase,
+  assertSingleIssue as assertSingleIssueBase,
+} from "../../test-utils";
+import { validateUnknownTagName } from "../../../src/validators/elements/unknown-tag-name";
+import {
+  getMessage,
+  UNKNOWN_CLASS_IN_NS,
+  UNKNOWN_CLASS_WITHOUT_NS,
+  UNKNOWN_TAG_NAME_IN_CLASS,
+  UNKNOWN_AGGREGATION_IN_CLASS,
+} from "../../../src/utils/messages";
+
+describe("the unknown tag name validation", () => {
+  let ui5SemanticModel: UI5SemanticModel;
+
+  before(async () => {
+    ui5SemanticModel = await generateModel({ version: "1.74.0" });
+  });
+
+  context("true positive scenarios", () => {
+    let assertSingleIssue: (xmlSnippet: string, message: string) => void;
+    before(() => {
+      assertSingleIssue = partial(
+        assertSingleIssueBase,
+        ui5SemanticModel,
+        {
+          element: [validateUnknownTagName],
+        },
+        "UnknownTagName",
+        "error"
+      );
+    });
+
+    context("tag with namespace", () => {
+      it("will detect an invalid class name in root tag", () => {
+        assertSingleIssue(
+          `<🢂mvc:View_TYPO🢀
+            xmlns:mvc="sap.ui.core.mvc"
+            xmlns="sap.m">
+          </mvc:View_TYPO>`,
+          getMessage(UNKNOWN_CLASS_IN_NS, "View_TYPO", "sap.ui.core.mvc")
```

Comment:
We are not actually testing the error message if we use the same logic to build it as it is built in production.

Reply:
And unlike "snapshot tests" no human is visually confirming it.

Reply:
This is true (we only test the parameters here, assuming they appear in the message string) but the other option is to have the message strings copied in the tests. The snapshot test has all the messages (unless I missed something) so in my opinion it's enough validation for the message strings.